### PR TITLE
Minor improvement on swath sampler

### PR DESCRIPTION
**Pull request by metdyn**
## Description

Minor changes to swath grid
- Give informative error message when swath grid Epoch does not equal swath sampler frequency
- Duplicated/Adapted subroutine RegridVector from GriddedIO.F90 to MAPL_EpochSwathMod.F90 (changing class name for ::this) because of the need to regrid and accumulate for swath sampler

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [x] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Review by tclune — COMMENTED** [review]
Do these changes actually do the tangent --> cartesian --> tangent needed for vectors?  Or just some of the initial work to populate the bundle?   If the answer is yes to the former, then please point me to that because I did not see it.

**Comment by metdyn** [comment]
It took me a while to find the overload in AbstractRegridder.   
Indeed, the tangent --> cartesian --> tangent is used as in 
https://github.com/GEOS-ESM/MAPL/blob/feature/ygyu/sampler_aist_test/base/MAPL_EsmfRegridder.F90#L562-L595

The code goes through MAPL_EsmfRegridder module via
https://github.com/GEOS-ESM/MAPL/blob/feature/ygyu/sampler_aist_test/base/NewRegridderManager.F90#L163-L164

I print out an output that shows
 TestDriver Date: 2019/08/01  Time: 00:40:00    0.0%Memory Committed
 regrid_vector_3d_real32


My statement the PR description is truly misleading and I have corrected it.  I did not add any regridding subroutines, all the functions are already in the MAPL, i.e., The griddedio.F90 code is here 
https://github.com/GEOS-ESM/MAPL/blob/feature/ygyu/sampler_aist_test/griddedio/GriddedIO.F90#L670-L835. 

I adapted the regridvector code for the swath sampler, because of the need to regrid and accumulate the interpolated values along the swath axis (only changed one line for the class name of this)

**Comment by metdyn** [comment]
I have built and run GEOS with this feature branch with C180 resolution

File: gridcomps/History/Sampler/MAPL_EpochSwathMod.F90 [hunk 1]
```diff
@@ -193,6 +193,39 @@ subroutine config_accumulate (this, key, cf, rc)
...
+    local_swath_epoch_sec = MAPL_nsecf( time_integer )
+
+    con = (hq_epoch_sec == local_swath_epoch_sec) .AND. (hq_epoch_sec == freq_sec)
+    if (mapl_am_i_root()) then
+       if (.not. con) then
+          write(6, '(2x,a,2x,i10)') 'hq_epoch_sec', hq_epoch_sec
+          write(6, '(2x,a,2x,i10)') 'local_swath_epoch_sec', local_swath_epoch_sec
+          write(6, '(2x,a,2x,i10)') 'freq_sec', freq_sec
+       end if
+    end if
```
Comment (tclune) [inline gridcomps/History/Sampler/MAPL_EpochSwathMod.F90:223]: Not in this pass, but please follow up with a PR to replace `write` with pFlogger calls.

**Review by tclune — APPROVED** [review]

**Pull request merged**